src/backend/joanie/lms_handler/backends/openedx.py

```diff
@@ -115,7 +115,7 @@ def set_enrollment(self, enrollment):
             if Order.objects.filter(
                 Q(target_courses=enrollment.course_run.course)
                 | Q(enrollment=enrollment),
-                state=enums.ORDER_STATE_VALIDATED,
+                state=enums.ORDER_STATE_COMPLETED,
```

Comment:
It should include all state that allow a user to enroll to a course run. Maybe we could create a constant `ORDER_STATE_ALLOW_ENROLLMENT`

```suggestion
                state__in=[
                    enums.ORDER_STATE_COMPLETED,
                    enums.ORDER_STATE_PENDING_PAYMENT,
                    enums.ORDER_STATE_FAILED_PAYMENT
                ]
```

Reply:
TODO added